src/detection_mac.cpp

```diff
@@ -459,3 +390,67 @@ void EIO_Find(uv_work_t* req) {
 
 	CreateFilteredList(&data->results, data->vid, data->pid);
 }
+
+static void WaitForDeviceHandled() {
+	uv_mutex_lock(&notify_mutex);
+	if(deviceHandled == false) {
+		uv_cond_wait(&notifyDeviceHandled, &notify_mutex);
+	}
+	deviceHandled = false;
+	uv_mutex_unlock(&notify_mutex);
+}
+
+static void SignalDeviceHandled() {
+	uv_mutex_lock(&notify_mutex);
+	deviceHandled = true;
+	uv_cond_signal(&notifyDeviceHandled);
+	uv_mutex_unlock(&notify_mutex);
+}
+
+static void cbWork(uv_work_t *req) {
+	uv_signal_start(&int_signal, cbTerm, SIGINT);
+	uv_signal_start(&term_signal, cbTerm, SIGTERM);
+
+	runLoopSource = IONotificationPortGetRunLoopSource(gNotifyPort);
+
+	gRunLoop = CFRunLoopGetCurrent();
+	CFRunLoopAddSource(gRunLoop, runLoopSource, kCFRunLoopDefaultMode);
+
+	// Start the run loop. Now we'll receive notifications.
+	CFRunLoopRun();
+
+	// We should never get here
+	fprintf(stderr, "Unexpectedly back from CFRunLoopRun()!\n");
+}
+
+static void cbAfter(uv_work_t *req, int status) {
+	uv_signal_stop(&int_signal);
+	uv_signal_stop(&term_signal);
+	uv_close((uv_handle_t *) &async_handler, NULL);
+	uv_cond_destroy(&notifyDeviceHandled);
+	uv_mutex_destroy(&notify_mutex);
+	CFRunLoopStop(CFRunLoopGetCurrent());
```

Comment:
I'm not confident in this `CFRunLoopStop` call here (kinda guessing on whether it is necessary)

Reply:
Should be `CFRunLoopStop(gRunLoop);`